pkg/aws/eni/node.go

```diff
@@ -290,15 +290,37 @@ func (n *Node) ResourceCopy() *v2.CiliumNode {
 	return n.resource.DeepCopy()
 }
 
-func (n *Node) getSecurityGroups() (securityGroups []string) {
-	// When no security groups are provided, derive them from eth0
-	securityGroups = n.resource.Spec.ENI.SecurityGroups
-	if len(securityGroups) == 0 {
-		if eni := n.manager.instancesAPI.GetENI(n.resource.Spec.ENI.InstanceID, 0); eni != nil {
-			securityGroups = eni.SecurityGroups
+func (n *Node) getSecurityGroupIDs(ctx context.Context) ([]string, error) {
+	// 1. check explicit security groups associations via checking Spec.ENI.SecurityGroups
+	// 2. check if Spec.ENI.SecurityGroupTags is passed and if so filter by those
+	// 3. if 1 and 2 give no results derive the security groups from eth0
+
+	eniSpec := n.resource.Spec.ENI
+	if len(eniSpec.SecurityGroups) > 0 {
+		return eniSpec.SecurityGroups, nil
+	}
+
+	if len(eniSpec.SecurityGroupTags) > 0 {
+		securityGroups := n.manager.instancesAPI.FindSecurityGroupByTags(eniSpec.VpcID, eniSpec.SecurityGroupTags)
```

Comment:
For the future:
Given this is potentially quite expensive, we could consider caching this until the security groups map is fetched the next time.

Reply:
Isn't the lookup locally anyway?

 From what I know the `InstancesManager` resync function gets called in the `eni-refresh` controller which has a 1m sync period. Do you mean that we should cache this info and not reach out to AWS every 1m for this info?

Reply:
@tgraf ping on the above ^^

Reply:
The main point was that I've seen envs with multiple 10K security groups so iterating for every new pod might be expensive. I think it's fine for a first rev though.

Longer term, we could improve the resync as well.

Reply:
Nothing blocking from my side for this PR.